src/System.Console/tests/WindowAndCursorProps.cs

```diff
@@ -0,0 +1,109 @@
+    public static void WindowWidth()
+    {
+        Assert.Throws<PlatformNotSupportedException>(() => Console.WindowWidth = 100);
+
+        // Validate that Console.WindowWidth returns some value in a non-redirected o/p.
+        RunInNonRedirectedOutput((data) => Console.WriteLine(Console.WindowWidth));
+        RunInRedirectedOutput((data) => Console.WriteLine(Console.WindowWidth));
+    }
+
+    //[Fact] //CI system makes it difficult to run things in a non-redirected environments.
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public static void NonRedirectedCursorVisible()
+    {
+        // Validate that Console.CursorVisible adds something to the stream when in a non-redirected environment.
+        RunInNonRedirectedOutput((data) => { Console.CursorVisible = true; Assert.True(data.ToArray().Length > 0); });
+        RunInNonRedirectedOutput((data) => { Console.CursorVisible = false; Assert.True(data.ToArray().Length > 0); });
+    }
+
+    [Fact]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public static void CursorVisible()
+    {
+        Assert.Throws<PlatformNotSupportedException>(() => { bool unused = Console.CursorVisible; });
+
+        // Validate that the Console.CursorVisible does nothing in a redirected stream.
+        RunInRedirectedOutput((data) => { Console.CursorVisible = true; Assert.Equal(0, data.ToArray().Length); });
+        RunInRedirectedOutput((data) => { Console.CursorVisible = false; Assert.Equal(0, data.ToArray().Length); });
+    }
+
+    internal static MemoryStream RunInRedirectedOutput(Action<MemoryStream> command)
+    {
+        // Make sure that redirecting to a memory stream causes no special writing to the stream when using Console.CursorVisible
+        MemoryStream data = new MemoryStream();
+        TextWriter savedOut = Console.Out;
+        try
+        {
+            Console.SetOut(new StreamWriter(data, new UTF8Encoding(false), 0x1000, leaveOpen: true) { AutoFlush = true });
+            command(data);
+        }
+        finally
+        {
+            Console.SetOut(savedOut);
+        }
+
+        return data;
+    }
+
+    internal static MemoryStream RunInNonRedirectedOutput(Action<MemoryStream> command)
+    {
+        // Make sure that when writing out to a UnixConsoleStream
+        // written out.
+        MemoryStream data = new MemoryStream();
+        TextWriter savedOut = Console.Out;
+        try
+        {
+            Console.SetOut(
+                new InterceptStreamWriter(
+                    Console.OpenStandardOutput(),
+                    new StreamWriter(data, new UTF8Encoding(false), 0x1000, leaveOpen: true) { AutoFlush = true },
+                    new UTF8Encoding(false), 0x1000, leaveOpen: true)
+                { AutoFlush = true });
+            command(data);
+        }
+        finally
+        {
+            Console.SetOut(savedOut);
+        }
+
+        return data;
+    }
+
+    private sealed class InterceptStreamWriter : StreamWriter
```

Comment:
We have something like this elsewhere in the Console tests, right?  If so, could we consolidate to a single definition?  This could even move to Common\tests\ if we think it might be useful elsewhere at some point, too.